ckanext/saml2/admin/fresh_idp_metadata.py

```diff
@@ -0,0 +1,38 @@
+#! /usr/bin/env python
+
+import xml.etree.cElementTree as ET
+from datetime import datetime
+import requests
+from argparse import ArgumentParser
+
+
+parser = ArgumentParser()
+
+parser.add_argument(
+    '-url',
+    help='URL for metadata download'
+)
+parser.add_argument(
+    '-path',
+    help='Path to current metadata xml file'
+)
+
+args = parser.parse_args()
+
+tree = ET.ElementTree(file=args.path)
+root = tree.getroot().attrib
+validUntil = root['validUntil']
+datetime_object = datetime.strptime(validUntil, '%Y-%m-%dT%H:%M:%SZ')
+validUntil = datetime_object.date()
+leftDays = (validUntil - datetime.today().date()).days
+
+if leftDays <= 1:
+    try:
+        r = requests.get(args.url, stream=True, verify=False)
+    except requests.exceptions.RequestException as e:
+        print 'Error: {0}'.format(e)
+    if r.status_code == 200:
+        with open('nswidhub-test-idp-new.xml', 'wb') as f:
```

Comment:
Why is the filename hardcoded when the script takes a path argument? Path should point to the filename of the IdP metadata used by CKAN so should be able to read from and write to that location.